{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Leish_Aneuploidy_Multiomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from DataSelect import *\n",
    "import pandas as pd\n",
    "from scipy.stats import ttest_ind\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "ANALYSISFOLDER= \"/Data\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load 4'omics data\n",
    "os.chdir(ANALYSISFOLDER)\n",
    "    \n",
    "CNVfile = 'CNVs.txt'\n",
    "Somyfile= 'Ploidy.txt'\n",
    "Transcriptfile = 'RNA.txt'\n",
    "Proteinfile = 'Proteins.txt'\n",
    "snpdf = pd.read_csv('Ldonovani.vcf', index_col = 0, skiprows = 62, sep =\"\\t\")\n",
    "\n",
    "anno = pd.read_csv('fullanno.csv', index_col = 0, sep = \";\")\n",
    "\n",
    "snpdf[\"SNP_type\"] = snpdf[\"INFO\"].str.split('ANN=').map(lambda x: x[1]).str.split('|').map(lambda x: x[1])\n",
    "snpdf[\"SNP_impact\"] = snpdf[\"INFO\"].str.split('ANN=').map(lambda x: x[1]).str.split('|').map(lambda x: x[2])\n",
    "snpdf[\"Genename\"] = snpdf[\"INFO\"].str.split('ANN=').map(lambda x: x[1]).str.split('|').map(lambda x: x[3])\n",
    "snpdf[\"SNP_substitution_base\"] = snpdf[\"INFO\"].str.split('ANN=').map(lambda x: x[1]).str.split('|').map(lambda x: x[9])\n",
    "snpdf[\"SNP_substitution_amino\"] = snpdf[\"INFO\"].str.split('ANN=').map(lambda x: x[1]).str.split('|').map(lambda x: x[10])\n",
    "snpdf[\"SNP_pos_transcript\"] = snpdf[\"INFO\"].str.split('ANN=').map(lambda x: x[1]).str.split('|').map(lambda x: x[11])\n",
    "snpdf[\"SNP_pos_protein\"] = snpdf[\"INFO\"].str.split('ANN=').map(lambda x: x[1]).str.split('|').map(lambda x: x[13])\n",
    "\n",
    "transdict, samples = readomic(Transcriptfile)\n",
    "proteindict, samples = readomic(Proteinfile)\n",
    "ploidydict, samples = readomic(Somyfile)\n",
    "CNVdict, samples = readomic(CNVfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_853/2297374771.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  CNV2[colname][index] = calculate_somy(row, somy, colname)\n",
      "/tmp/ipykernel_853/2297374771.py:21: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  somypergene[colname][index] = somy[colname][chrom]\n"
     ]
    }
   ],
   "source": [
    "somy = pd.DataFrame.from_dict(ploidydict, orient='index', dtype = float)\n",
    "CNV = pd.DataFrame.from_dict(CNVdict, orient='index', dtype = float)\n",
    "CNV[\"chrom\"] = ['Ld' + x.split('_')[1][0:2] for x in list(CNV.index)]\n",
    "CNV2 = CNV.copy()\n",
    "somypergene = CNV.copy()\n",
    "\n",
    "def calculate_somy(row, somy, colname):\n",
    "    return row[colname] * somy[colname][row[\"chrom\"]]\n",
    "\n",
    "for colname in CNV2:\n",
    "    if colname != 'chrom':\n",
    "        if '.1' in colname:\n",
    "            for index, row in CNV2.iterrows():\n",
    "                CNV2[colname][index] = calculate_somy(row, somy, colname)\n",
    "\n",
    "for colname in somypergene:\n",
    "    if colname != 'chrom':\n",
    "        if '.1' in colname:\n",
    "            for index, row in somypergene.iterrows():\n",
    "                chrom = somypergene['chrom'][index]\n",
    "                somypergene[colname][index] = somy[colname][chrom]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h3> Define Samples </h3>\n",
    "\n",
    "    Define samples in first and second condition. Samples for transcript and proteins are typically the same, exept for BPK275, where one transcript replicate (275.4) was dropped for sequencing quality reasons."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "strains = {'275.1' : 'BPK275A1', '173.1' : 'BPK173A1', '282.1': 'BPK282B2', '575.1' : 'BHU575A1','288.1' : 'BPK288B2', '178.1' : 'BPK178A1'}\n",
    "\n",
    "combos = [\n",
    "    {\"STRAIN_A\" : ['173.1','173.2','173.3','173.4','173L.1', '173L.2'],\n",
    "    \"STRAIN_B\" : ['275.1','275.2','275.3','275L.1','275L.2'],\n",
    "    \"STRAIN_B_proteinlevel\" : ['275.1','275.2','275.3','275.4','275L.1','275L.2'],\n",
    "    \"outfileID\" : 'F_275vs173'},\n",
    "\n",
    "{\"STRAIN_A\" : ['288.1','288.2','288.3','288.4'],\n",
    "\"STRAIN_B\" : ['275.1','275.2','275.3'],\n",
    "\"STRAIN_B_proteinlevel\" : ['275.1','275.2','275.3','275.4'],\n",
    "\"outfileID\" : 'F_275vs288'},\n",
    "\n",
    "{\"STRAIN_A\" : ['288.1','288.2','288.3','288.4'],\n",
    "\"STRAIN_B\" : ['173.1','173.2','173.3','173.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['173.1','173.2','173.3','173.4'],\n",
    "\"outfileID\" : 'F_173vs288'},\n",
    "\n",
    "#\n",
    "    \n",
    "{\"STRAIN_A\" : ['575.1','575.2','575.3','575.4'],\n",
    "\"STRAIN_B\" : ['288.1','288.2','288.3','288.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['288.1','288.2','288.3','288.4'],\n",
    "\"outfileID\" : 'F_288vs575'},\n",
    "    \n",
    "{\"STRAIN_A\" : ['575.1','575.2','575.3','575.4'],\n",
    "\"STRAIN_B\" : ['275.1','275.2','275.3'],\n",
    "\"STRAIN_B_proteinlevel\" : ['275.1','275.2','275.3','275.4'],\n",
    "\"outfileID\" : 'F_275vs575'},\n",
    "    \n",
    "{\"STRAIN_A\" : ['575.1','575.2','575.3','575.4'],\n",
    "\"STRAIN_B\" : ['173.1','173.2','173.3','173.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['173.1','173.2','173.3','173.4'],\n",
    "\"outfileID\" : 'F_173vs575'},\n",
    "\n",
    "#\n",
    "\n",
    "{\"STRAIN_A\" : ['282.1','282.2','282.3','282.4'],\n",
    "\"STRAIN_B\" : ['275.1','275.2','275.3'],\n",
    "\"STRAIN_B_proteinlevel\" : ['275.1','275.2','275.3','275.4'],\n",
    "\"outfileID\" : 'F_275vs282'},\n",
    "\n",
    "{\"STRAIN_A\" : ['282.1','282.2','282.3','282.4'],\n",
    "\"STRAIN_B\" : ['575.1','575.2','575.3','575.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['575.1','575.2','575.3','575.4'],\n",
    "\"outfileID\" : 'F_575vs282'},\n",
    "\n",
    "{\"STRAIN_A\" : ['282.1','282.2','282.3','282.4'],\n",
    "\"STRAIN_B\" : ['288.1','288.2','288.3','288.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['288.1','288.2','288.3','288.4'],\n",
    "\"outfileID\" : 'F_288vs282'},\n",
    "    \n",
    "{\"STRAIN_A\" : ['282.1','282.2','282.3','282.4'],\n",
    "\"STRAIN_B\" : ['173.1','173.2','173.3','173.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['173.1','173.2','173.3','173.4'],\n",
    "\"outfileID\" : 'F_173vs282'},\n",
    "\n",
    "####################################################\n",
    "\n",
    "{\"STRAIN_A\" : ['178.1','178.2','178.3','178.4','178L.1', '178L.2'],\n",
    "\"STRAIN_B\" : ['275.1','275.2','275.3','275L.1','275L.2'],\n",
    "\"STRAIN_B_proteinlevel\" : ['275.1','275.2','275.3','275.4','275L.1','275L.2'],\n",
    "\"outfileID\" : 'F_275vs178'},\n",
    "    \n",
    "{\"STRAIN_A\" : ['178.1','178.2','178.3','178.4'],\n",
    "\"STRAIN_B\" : ['288.1','288.2','288.3','288.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['288.1','288.2','288.3','288.4'],\n",
    "\"outfileID\" : 'F_288vs178'},\n",
    "    \n",
    "{\"STRAIN_A\" : ['178.1','178.2','178.3','178.4'],\n",
    "\"STRAIN_B\" : ['282.1','282.2','282.3','282.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['178.1','178.2','178.3','178.4'],\n",
    "\"outfileID\" : 'F_178vs282'},\n",
    "\n",
    "{\"STRAIN_A\" : ['178.1','178.2','178.3','178.4'],\n",
    "\"STRAIN_B\" : ['575.1','575.2','575.3','575.4'],\n",
    "\"STRAIN_B_proteinlevel\" : ['575.1','575.2','575.3','575.4'],\n",
    "\"outfileID\" : 'F_575vs178'},\n",
    "\n",
    "{\"STRAIN_A\" : ['178.1','178.2','178.3','178.4','178L.1', '178L.2'],\n",
    "\"STRAIN_B\" : ['173.1','173.2','173.3','173.4','173L.1', '173L.2'],\n",
    "\"STRAIN_B_proteinlevel\" : ['173.1','173.2','173.3','173.4','173L.1', '173L.2'],\n",
    "\"outfileID\" : 'F_173vs178'}\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summarydict = {}\n",
    "transcriptpds = []\n",
    "proteinpds = []\n",
    "FDR = 0.05\n",
    "\n",
    "for i in combos:\n",
    "    strain_A = i[\"STRAIN_A\"][0]\n",
    "    strain_B = i[\"STRAIN_B\"][0]\n",
    "    strain_A_replicates = i[\"STRAIN_A\"]\n",
    "    strain_B_replicates = i[\"STRAIN_B\"]\n",
    "    outfileID = i[\"outfileID\"]\n",
    "    strain_B_protein = i[\"STRAIN_B_proteinlevel\"]\n",
    "    print(outfileID)\n",
    "\n",
    "    ###\n",
    "    ###GENOME\n",
    "    ###\n",
    "\n",
    "    Dosagetotal = (CNV2[strain_B]/CNV2[strain_A]).to_frame()\n",
    "    Dosagetotal = np.log2(Dosagetotal)\n",
    "    Dosagetotal.index.name = 'GeneName'\n",
    "    Dosagetotal.columns = ['Dosage_Log2FC']\n",
    "\n",
    "    CNVdif = (CNV[strain_B]/CNV[strain_A]).to_frame()\n",
    "    CNVdif = np.log2(CNVdif)\n",
    "    CNVdif.index.name = 'GeneName'\n",
    "    CNVdif.columns = ['CNV_Log2FC']\n",
    "\n",
    "    somypergenedif = (somypergene[strain_B]/somypergene[strain_A]).to_frame()\n",
    "    somypergenedif = np.log2(somypergenedif)\n",
    "    somypergenedif.index.name = 'GeneName'\n",
    "    somypergenedif.columns = ['Somy_Log2FC']\n",
    "\n",
    "    ###\n",
    "    ###TRANSCRIPTOME\n",
    "    ###\n",
    "    \n",
    "    ##\n",
    "    ##Normalisation\n",
    "    ##\n",
    "    \n",
    "    log2FC_Trans = pd.DataFrame()\n",
    "    df = pd.DataFrame.from_dict(transdict, orient='index', dtype = float)\n",
    "    df = df.drop(['275.4'], axis=1)\n",
    "    #add pseudocounts for remaining zeros\n",
    "    df = df.replace(0, 1)\n",
    "    #take 2 log\n",
    "    df = np.log2(df)\n",
    "    #divide column by column mean\n",
    "    for column in df:\n",
    "        df[column] = df[column] - np.average(df[column])\n",
    "       \n",
    "    ##\n",
    "    ##Calculate Fold Change\n",
    "    ##\n",
    "    \n",
    "    denominator = df[strain_A_replicates]\n",
    "    numerator = df[strain_B_replicates]\n",
    "    log2FC = numerator.mean(axis = 1) - denominator.mean(axis = 1)\n",
    "    log2FC = log2FC.to_frame('transcript_log2fc')\n",
    "    for index, row in numerator.iterrows():\n",
    "        log2FC.loc[index,'transcript_pval'] = ttest_ind(numerator.loc[index], denominator.loc[index])[1]\n",
    "    log2FC_Trans = log2FC_Trans.append(log2FC)\n",
    "    log2FC_Trans.index.name ='GeneName'\n",
    "    log2FC_Trans[\"GeneName2\"] = log2FC_Trans.index\n",
    "\n",
    "    ###\n",
    "    ###PROTEOME\n",
    "    ###\n",
    "    \n",
    "    ##\n",
    "    ##Normalisation\n",
    "    ##\n",
    "\n",
    "    df = pd.DataFrame.from_dict(proteindict, orient='index', dtype = float)\n",
    "    df = df.replace(0, np.nan)\n",
    "    df.describe()\n",
    "    normprot = df\n",
    "    \n",
    "    ##\n",
    "    ##Calculate Fold Change\n",
    "    ##\n",
    "    \n",
    "    log2FC_Prot = pd.DataFrame()\n",
    "    denominator = np.log2(normprot[strain_A_replicates])\n",
    "    numerator = np.log2(normprot[strain_B_protein])\n",
    "    log2FC = numerator.mean(axis = 1) - denominator.mean(axis = 1)\n",
    "    log2FC = log2FC.to_frame('protein_log2fc')\n",
    "    for index, row in numerator.iterrows():\n",
    "        log2FC.loc[index,'pval_protein'] = ttest_ind(numerator.loc[index].dropna(), denominator.loc[index].dropna())[1]\n",
    "        log2FC.loc[index,'samplesizeSample_protein'] = len(numerator.loc[index].dropna())\n",
    "        log2FC.loc[index,'samplesizeControl_protein'] = len(denominator.loc[index].dropna())\n",
    "        log2FC.loc[index,'nstdev_sample'] = numerator.loc[index].dropna().std()/numerator.loc[index].dropna().mean()\n",
    "        log2FC.loc[index,'nstdev_control'] = denominator.loc[index].dropna().std()/denominator.loc[index].dropna().mean()\n",
    "    log2FC_Prot = log2FC_Prot.append(log2FC)\n",
    "    log2FC_Prot.index.name ='GeneName'\n",
    "    \n",
    "    ###\n",
    "    ###INTEGRATE\n",
    "    ###\n",
    "    \n",
    "    ##\n",
    "    ## Integrate\n",
    "    ##\n",
    "    \n",
    "    result = pd.merge(log2FC_Trans, log2FC_Prot, left_on='GeneName', right_on='GeneName',sort=True, how='left')\n",
    "    result = pd.merge(result, Dosagetotal, left_on='GeneName', right_on='GeneName',sort=True, how='left')\n",
    "    result = pd.merge(result, CNVdif, left_on='GeneName', right_on='GeneName',sort=True, how='left')\n",
    "    result = pd.merge(result, somypergenedif, left_on='GeneName', right_on='GeneName',sort=True, how='left')\n",
    "    result[\"compensation\"] = result[\"protein_log2fc\"] - result[\"transcript_log2fc\"]\n",
    "    result[\"Strains\"] = outfileID[2:]\n",
    "    result = result[(result ['CNV_Log2FC'] < 0.3) & (result ['CNV_Log2FC'] > -0.3)]\n",
    "\n",
    "    anno['LdBPKV2'] = anno['LdBPKV2'].str.replace('.1','', regex = False)\n",
    "\n",
    "    result = pd.merge(result, anno, left_on = \"GeneName\", right_on='LdBPKV2',sort=True, how='left').drop(columns=[\"Eccentricity\", \"IsSingleNode\",\n",
    "                                'ClosenessCentrality','ClusteringCoefficient',\"Stress\",\"Degree\",\n",
    "                                \"Radiality\",\"BetweennessCentrality\",\"PartnerOfMultiEdgedNodePairs\",\"NeighborhoodConnectivity\",\"Gravy\",\n",
    "                                 \"NumberOfDirectedEdges\",\"NumberOfUndirectedEdges\",\"AverageShortestPathLength\",\"TopologicalCoefficient\",\"degree\"])\n",
    "    result = result[~result.LdBPKV2.str.contains(\"LdBPK_3700\", na=False)]\n",
    "    result['SNPs'] = result.apply(lambda row: checkSNP(strains[strain_B],strains[strain_A], row[\"L. donovani\"], snpdf), axis=1)\n",
    "    proteinpds.append(result)\n",
    "    \n",
    "    ##\n",
    "    ## Calculate cis- en trans-transcripts\n",
    "    ##\n",
    "    log2FC_Trans_sel = result[result['transcript_pval'].notna()]\n",
    "    log2FC_Trans_sel['trans_padj'] = multipletests(log2FC_Trans_sel['transcript_pval'], alpha=0.05, method='fdr_bh', is_sorted=False, returnsorted=False)[1]\n",
    "    log2FC_Trans_sel[\"aneuploidy-driven\"] = (log2FC_Trans_sel['trans_padj'] < FDR) & (log2FC_Trans_sel['Somy_Log2FC'].abs() < 0.30)\n",
    "    cis = len(log2FC_Trans_sel[(log2FC_Trans_sel['trans_padj'] < FDR) & (log2FC_Trans_sel['Somy_Log2FC'].abs() > 0.30)])\n",
    "    trans = len(log2FC_Trans_sel[(log2FC_Trans_sel['trans_padj'] < FDR) & (log2FC_Trans_sel['Somy_Log2FC'].abs() < 0.30)])\n",
    "    total = float(cis) + float(trans)\n",
    "    if total != 0:\n",
    "        summarydict[outfileID[2:]] = {\"protein\" :0, \"transcript\" : [cis,trans,total,(float(cis)/total*100.0), (float(trans)/total*100.0)]}\n",
    "    else:\n",
    "        summarydict[outfileID[2:]] = {\"protein\" : 0, \"transcript\" : [cis,trans,total,\"NA\",\"NA\"]}\n",
    "    log2FC_Trans_sel.index.name ='GeneName'\n",
    "    transcriptpds.append(log2FC_Trans_sel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.concat(proteinpds, axis=0).to_csv('allcomp_final.csv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:76: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisdown['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:75: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cisup['Cis/Trans'] = \"Cis\"\n",
      "/tmp/ipykernel_853/3562603053.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transup['Cis/Trans'] = \"Trans\"\n",
      "/tmp/ipykernel_853/3562603053.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  transdown['Cis/Trans'] = \"Trans\"\n"
     ]
    }
   ],
   "source": [
    "FDR = 0.05\n",
    "samplesize = 3\n",
    "FC = 0.0\n",
    "\n",
    "Differential_Proteins = []\n",
    "totaltrans = {}\n",
    "totalprot = {}\n",
    "\n",
    "dfObj = pd.DataFrame(columns=['Comparisson', 'Diff_Total','Diff_Cis','Diff_Trans','Direction', 'Proteins_All','Proteins_Cis','Proteins_Trans'])\n",
    "\n",
    "totaldif = 0\n",
    "newpds = []\n",
    "for i in range(len(proteinpds)):\n",
    "    strains = proteinpds[i].loc[1,\"Strains\"]\n",
    "    log2FC_Prot_sel = proteinpds[i]\n",
    "    log2FC_Prot_sel = log2FC_Prot_sel[log2FC_Prot_sel['samplesizeSample_protein'] > samplesize]\n",
    "    log2FC_Prot_sel = log2FC_Prot_sel[log2FC_Prot_sel['samplesizeControl_protein'] > samplesize]\n",
    "    log2FC_Prot_sel = log2FC_Prot_sel[log2FC_Prot_sel['pval_protein'].notna()]\n",
    "    log2FC_Prot_sel['prot_padj'] = multipletests(log2FC_Prot_sel['pval_protein'], alpha=0.05, method='fdr_bh', is_sorted=False, returnsorted=False)[1]\n",
    "    log2FC_Prot_sel[\"aneuploidy-driven\"] = (log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel['Somy_Log2FC'].abs() < 0.30)\n",
    "    log2FC_Prot_sel[\"L. major\"] =  log2FC_Prot_sel[\"L. major\"].str.split(',').str[0]\n",
    "    newpds.append(log2FC_Prot_sel)\n",
    "    \n",
    "# Need to make sure it matches the right L.donovani gene\n",
    "    difprot = log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel[\"protein_log2fc\"] > 0)][\"L. major\"].to_list()\n",
    "    up = len(difprot)\n",
    "    totaldif += len(difprot)\n",
    "    upproteins = log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel[\"protein_log2fc\"] > 0)][\"L. major\"].dropna().to_list()\n",
    "    difprot = log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel[\"protein_log2fc\"] < 0)][\"L. major\"].to_list()\n",
    "    down = len(difprot)\n",
    "    totaldif += len(difprot)\n",
    "    downproteins = log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel[\"protein_log2fc\"] < 0)][\"L. major\"].dropna().to_list()\n",
    "    \n",
    "    ##\n",
    "    ## Calculate cis- en trans-proteins\n",
    "    ##\n",
    "    \n",
    "    cis = len(log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel['Somy_Log2FC'].abs() >= 0.30)])\n",
    "    trans = len(log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel['Somy_Log2FC'].abs() < 0.30)])\n",
    "    total = float(cis) + float(trans)\n",
    "    if total != 0:\n",
    "        summarydict[strains]['protein'] = [cis,trans,total,(float(cis)/total*100.0), (float(trans)/total*100.0)]\n",
    "    else:\n",
    "        summarydict[strains]['protein'] = [cis,trans,total,\"NA\",\"NA\"]\n",
    "        \n",
    "    ##\n",
    "    ## Determine cis- en trans-proteins up-and down\n",
    "    ##\n",
    "    \n",
    "    log2FC_Prot_sel = log2FC_Prot_sel.sort_values(by=['Somy_Log2FC'], ascending=False)\n",
    "    for prot in log2FC_Prot_sel[\"L. major\"].dropna().to_list():\n",
    "        totalprot[prot] = 0\n",
    "\n",
    "    cisup = log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel['Somy_Log2FC'].abs() >= 0.30) & (log2FC_Prot_sel[\"protein_log2fc\"] >= FC)]\n",
    "    transup = log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel['Somy_Log2FC'].abs() < 0.30) & (log2FC_Prot_sel[\"protein_log2fc\"] >= FC)]\n",
    "    cisdown = log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel['Somy_Log2FC'].abs() >= 0.30) & (log2FC_Prot_sel[\"protein_log2fc\"] < -FC)]\n",
    "    transdown = log2FC_Prot_sel[(log2FC_Prot_sel['prot_padj'] < FDR) & (log2FC_Prot_sel['Somy_Log2FC'].abs() < 0.30) & (log2FC_Prot_sel[\"protein_log2fc\"] < -FC)]\n",
    "    allup = pd.concat([cisup, transup], ignore_index=True)\n",
    "    alldown = pd.concat([cisdown, transdown], ignore_index=True)\n",
    "    alltrans = pd.concat([transup, transdown], ignore_index=True)\n",
    "    \n",
    "    for prot in alltrans[\"L. major\"].dropna().to_list():\n",
    "        totaltrans[prot] = 0\n",
    "    \n",
    "    cisup_lmaj = cisup[\"L. major\"].dropna().to_list()\n",
    "    transup_lmaj = transup[\"L. major\"].dropna().to_list()\n",
    "    cisdown_lmaj = cisdown[\"L. major\"].dropna().to_list()\n",
    "    transdown_lmaj = transdown[\"L. major\"].dropna().to_list()\n",
    "    allup_lmaj = pd.concat([cisup, transup], ignore_index=True)[\"L. major\"].dropna().to_list()\n",
    "    alldown_lmaj = pd.concat([cisdown, transdown], ignore_index=True)[\"L. major\"].dropna().to_list()\n",
    "    \n",
    "    dfObj = dfObj.append({'Comparisson' : strains, 'Diff_Total' : len(allup_lmaj), 'Diff_Cis' : len(cisup_lmaj), 'Diff_Trans' : len(transup_lmaj) , 'Direction' : 'Up', 'Proteins_All' : \" \".join([i.split(',')[0] for i in allup_lmaj]),'Proteins_Cis' : \" \".join([i.split(',')[0] for i in cisup_lmaj]), 'Proteins_Trans' : \" \".join([i.split(',')[0] for i in transup_lmaj])}, ignore_index=True)\n",
    "    dfObj = dfObj.append({'Comparisson' : strains, 'Diff_Total' : len(alldown_lmaj), 'Diff_Cis' : len(cisdown_lmaj), 'Diff_Trans' : len(transdown_lmaj), 'Direction' : 'Down', 'Proteins_All' : \" \".join([i.split(',')[0] for i in alldown_lmaj]),'Proteins_Cis' : \" \".join([i.split(',')[0] for i in cisdown_lmaj]), 'Proteins_Trans' : \" \".join([i.split(',')[0] for i in transdown_lmaj])}, ignore_index=True)\n",
    "    \n",
    "    cisup['Cis/Trans'] = \"Cis\"\n",
    "    cisdown['Cis/Trans'] = \"Cis\"\n",
    "    transup['Cis/Trans'] = \"Trans\"\n",
    "    transdown['Cis/Trans'] = \"Trans\"\n",
    "    Differential_Proteins.append(pd.concat([cisup,cisdown,transup,transdown]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.concat(Differential_Proteins).to_csv('CisTrans_final.csv', sep = ';')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Table S1.H Trans-Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(203, 5)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat(Differential_Proteins)\n",
    "df = df[df[\"Cis/Trans\"] == \"Trans\"]\n",
    "df.to_csv('test.csv', sep = \";\")\n",
    "df['Duplicated'] = df['GeneName2'].duplicated(keep = False)\n",
    "df['In_Comparissons'] = df.groupby('GeneName2')['Strains'].transform(lambda x : ', '.join(x))\n",
    "df = df[[\"GeneName2\",\"GeneID\",\"Duplicated\",\"In_Comparissons\",\"SNPs\", \"anno\"]]\n",
    "df = df.set_index(\"GeneName2\")\n",
    "df['SNPs'] = df['SNPs'].astype(str)\n",
    "df = df.drop_duplicates()\n",
    "df.to_csv('Trans_Proteins_final.csv', sep = ';')\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Table S1.G Trans-Transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(541, 5)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat(transcriptpds)\n",
    "df = df[df[\"aneuploidy-driven\"] == False]\n",
    "df = df[df[\"trans_padj\"] < FDR]\n",
    "df['Duplicated'] = df['GeneName2'].duplicated(keep = False)\n",
    "df['In_Comparissons'] = df.groupby('GeneName2')['Strains'].transform(lambda x : ', '.join(x))\n",
    "df = df[[\"GeneName2\",\"GeneID\",\"Duplicated\",\"In_Comparissons\",\"SNPs\", \"anno\"]]\n",
    "df = df.set_index(\"GeneName2\")\n",
    "df['SNPs'] = df['SNPs'].astype(str)\n",
    "df = df.drop_duplicates()\n",
    "df.to_csv('Trans_Transcripts_final.csv', sep = ';')\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Table S1.F Differential Transcripts & Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "275vs173\t86.56716417910447\t13.432835820895523\t134.0\t40.65934065934066\t59.34065934065934\t91.0\n",
      "275vs288\t90.9090909090909\t9.090909090909092\t11.0\t48.0\t52.0\t125.0\n",
      "173vs288\t86.25\t13.750000000000002\t80.0\t55.55555555555556\t44.44444444444444\t27.0\n",
      "288vs575\t65.75342465753424\t34.24657534246575\t73.0\t71.42857142857143\t28.57142857142857\t14.0\n",
      "275vs575\t84.82758620689656\t15.172413793103448\t145.0\t58.92857142857143\t41.07142857142857\t56.0\n",
      "173vs575\tNA\tNA\t0.0\t50.0\t50.0\t4.0\n",
      "275vs282\t77.61194029850746\t22.388059701492537\t67.0\t46.42857142857143\t53.57142857142857\t84.0\n",
      "575vs282\t41.17647058823529\t58.82352941176471\t34.0\t6.25\t93.75\t16.0\n",
      "288vs282\t60.86956521739131\t39.130434782608695\t23.0\t39.34426229508197\t60.65573770491803\t61.0\n",
      "173vs282\t74.35897435897436\t25.64102564102564\t39.0\t48.148148148148145\t51.85185185185185\t27.0\n",
      "275vs178\t79.79274611398964\t20.207253886010363\t386.0\t44.761904761904766\t55.23809523809524\t105.0\n",
      "288vs178\t54.54545454545454\t45.45454545454545\t11.0\t45.45454545454545\t54.54545454545454\t11.0\n",
      "178vs282\t71.42857142857143\t28.57142857142857\t7.0\tNA\tNA\t0.0\n",
      "575vs178\t0.0\t100.0\t2.0\tNA\tNA\t0.0\n",
      "173vs178\t55.00000000000001\t45.0\t40.0\t26.31578947368421\t73.68421052631578\t19.0\n"
     ]
    }
   ],
   "source": [
    "for i in summarydict:\n",
    "    print(\"\\t\".join([str(j) for j in [i, summarydict[i]['transcript'][3],summarydict[i]['transcript'][4],summarydict[i]['transcript'][2],summarydict[i]['protein'][3],summarydict[i]['protein'][4],summarydict[i]['protein'][2]]]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
